src/abstractdataframe/iteration.jl

```diff
@@ -143,39 +139,39 @@ julia> sum.(eachcol(df))
 2-element Array{Int64,1}:
  10
  50
-
-julia> collect(eachcol(df, true))
-2-element Array{Pair{Symbol,AbstractArray{T,1} where T},1}:
- :x => [1, 2, 3, 4]
- :y => [11, 12, 13, 14]
 ```
 """
-@inline function eachcol(df::T, names::Bool=false) where T<: AbstractDataFrame
-    if names
-        DataFrameColumns{T, Pair{Symbol, AbstractVector}}(df)
-    else
-        DataFrameColumns{T, AbstractVector}(df)
-    end
-end
+eachcol(df::AbstractDataFrame) = DataFrameColumns(df)
 
 Base.size(itr::DataFrameColumns) = (size(parent(itr), 2),)
 Base.IndexStyle(::Type{<:DataFrameColumns}) = Base.IndexLinear()
 
-@inline function Base.getindex(itr::DataFrameColumns{<:AbstractDataFrame,
-                                                     Pair{Symbol, AbstractVector}}, j::Int)
-    @boundscheck checkbounds(itr, j)
-    @inbounds _names(parent(itr))[j] => parent(itr)[!, j]
-end
-
-@inline function Base.getindex(itr::DataFrameColumns{<:AbstractDataFrame, AbstractVector}, j::Int)
+@inline function Base.getindex(itr::DataFrameColumns, j::Int)
     @boundscheck checkbounds(itr, j)
     @inbounds parent(itr)[!, j]
 end
 
+Base.getindex(itr::DataFrameColumns, j::Symbol) = parent(itr)[!, j]
+
 Base.getproperty(itr::DataFrameColumns, col_ind::Symbol) = getproperty(parent(itr), col_ind)
 # Private fields are never exposed since they can conflict with column names
 Base.propertynames(itr::DataFrameColumns, private::Bool=false) = propertynames(parent(itr))
 
+"""
+    keys(::DataFrameColumns)
+
+Get a vector of column names of a parent data frame.
```

Comment:
Do we really need to mention the parent?
```suggestion
Get a vector of column names of `dfc`.
```